eventsource-client/src/client.rs

```diff
@@ -517,30 +517,49 @@ where
                             }
                         }
 
-                        self.as_mut().reset_redirects();
-                        self.as_mut().project().state.set(State::New);
-
-                        return Poll::Ready(Some(Err(Error::UnexpectedResponse(
+                        let error = Error::UnexpectedResponse(
```

Comment:
Instead of introducing the new error variant, I am using `UnexpectedResponse`. This matches the error that would be returned if we were retrying as well.